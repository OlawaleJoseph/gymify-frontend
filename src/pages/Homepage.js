import React from 'react';
import { Link, Redirect } from 'react-router-dom';
import BgImage from '../assets/images/background.jpg';
import { getHeaders } from '../utils/common';

const Homepage = () => {
  const headers = getHeaders();
  if (headers && Object.keys(headers).length) return <Redirect to="/dashboard" />;

  return (
    <div className="homepage">
      <nav className="d-flex px-5 justify-content-between align-items-center homepage__nav">
        <Link className="logo" to="/">GYMIFY</Link>
        <div className="d-flex">
          <Link className="mr-4 btn btn-info" to="/register">Register</Link>
          <Link className="btn btn-info" to="/login">Login</Link>
        </div>
      </nav>
      <main className="px-5 d-flex justify-content-start align-items-center homepage__main" style={{ backgroundImage: `url(${BgImage})` }}>
        <div className="homepage__welcome_text">
          <h1>We Never Stop! We never falter!</h1>
          <h3>Keep fit with pride</h3>
          <Link className="btn btn-primary btn-lg" to="/register">Register</Link>
        </div>
      </main>
    </div>
  );
};

export default Homepage;
