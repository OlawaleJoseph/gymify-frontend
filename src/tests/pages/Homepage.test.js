import { Link } from 'react-router-dom';
import { shallowRenderWithoutProvider } from '../setup';
import Homepage from '../../pages/Homepage';

describe('Homepage', () => {
  const page = shallowRenderWithoutProvider(Homepage);

  test('Should render Homepage', () => {
    expect(page).toBeTruthy();
    expect(page).not.toBeFalsy();
    expect(page).toMatchSnapshot();
  });

  test('Should render nav', () => {
    const nav = page.find('nav');

    expect(nav.length).toBe(1);
    expect(nav.length).not.toBe(0);
  });

  test('Should render logo', () => {
    const logo = page.find('.logo');

    expect(logo.length).toBe(1);
    expect(logo.length).not.toBe(0);
    expect(logo.text()).not.toBe('');
    expect(logo.text()).toBe('GYMIFY');
  });

  test('Should render register and login links in nav', () => {
    const nav = page.find('nav div');
    const links = nav.find(Link);

    expect(links.length).toBe(2);
    expect(nav.length).not.toBe(0);
    expect(links.first().text()).toBe('Register');
    expect(links.last().text()).toBe('Login');
  });

  test('Should render welcome text', () => {
    const h1 = page.find('h1');
    const h3 = page.find('h3');

    expect(h1.length).toBe(1);
    expect(h1.length).not.toBe(0);
    expect(h3.length).toBe(1);
    expect(h3.length).not.toBe(0);
    expect(h1.text()).toBe('We Never Stop! We never falter!');
    expect(h1.text()).not.toBe('');
    expect(h3.text()).toBe('Keep fit with pride');
    expect(h3.text()).not.toBe('');
  });
});
