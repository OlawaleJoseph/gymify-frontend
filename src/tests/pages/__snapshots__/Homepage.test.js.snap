// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Should render Homepage 1`] = `ShallowWrapper {}`;
